WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderBlogFragment.java

```diff
@@ -83,39 +97,78 @@ private void checkEmptyView() {
     public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
         mRecyclerView.setAdapter(getBlogAdapter());
-        refresh();
     }
 
     @Override
     public void onSaveInstanceState(Bundle outState) {
         outState.putSerializable(ARG_BLOG_TYPE, getBlogType());
-        outState.putBoolean(ReaderConstants.KEY_WAS_PAUSED, mWasPaused);
+        if (getBlogAdapter().hasSearchFilter()) {
+            outState.putString(KEY_SEARCH_FILTER, getBlogAdapter().getSearchFilter());
+        }
         super.onSaveInstanceState(outState);
     }
 
     private void restoreState(Bundle args) {
         if (args != null) {
-            mWasPaused = args.getBoolean(ReaderConstants.KEY_WAS_PAUSED);
             if (args.containsKey(ARG_BLOG_TYPE)) {
                 mBlogType = (ReaderBlogType) args.getSerializable(ARG_BLOG_TYPE);
             }
+            if (args.containsKey(KEY_SEARCH_FILTER)) {
+                mSearchFilter = args.getString(KEY_SEARCH_FILTER);
+            }
         }
     }
 
     @Override
-    public void onPause() {
-        super.onPause();
-        mWasPaused = true;
+    public void onResume() {
+        super.onResume();
+        refresh();
     }
 
+    /*
+     * note this will only be called for followed blogs
+     */
     @Override
-    public void onResume() {
-        super.onResume();
-        // refresh the adapter if the fragment is resuming from a paused state so that changes
-        // made in another activity (such as follow state) are reflected here
-        if (mWasPaused) {
-            mWasPaused = false;
-            refresh();
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        inflater.inflate(R.menu.reader_subs, menu);
+
+        MenuItem searchMenu = menu.findItem(R.id.menu_search);
+        SearchView searchView = (SearchView) searchMenu.getActionView();
+        searchView.setQueryHint(getString(R.string.reader_hint_search_followed_sites));
+
+        MenuItemCompat.setOnActionExpandListener(searchMenu, new MenuItemCompat.OnActionExpandListener() {
+            @Override
+            public boolean onMenuItemActionExpand(MenuItem item) {
+                return true;
+            }
+            @Override
+            public boolean onMenuItemActionCollapse(MenuItem item) {
+                setSearchFilter(null);
+                return true;
+            }
+        });
+
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                setSearchFilter(query);
+                return false;
+            }
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                if (!TextUtils.isEmpty(newText)) {
```

Comment:
Oh that makes a lot sense. It will require some hacky way to prevent this as you stated which probably is not worth it. We probably should open an issue anyway even if for other devs to see if they wonder why that's happening. We might also find a way later on to differentiate the user change from system change.